clustered/integration-test/src/test/java/org/ehcache/clustered/TerminatedServerTest.java

```diff
@@ -0,0 +1,132 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered;
+
+import org.ehcache.CacheManager;
+import org.ehcache.Diagnostics;
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.terracotta.testing.rules.BasicExternalCluster;
+import org.terracotta.testing.rules.Cluster;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Provides integration tests in which the server is terminated before the Ehcache operation completes.
+ */
+public class TerminatedServerTest {
+
+  private static final String RESOURCE_CONFIG =
+      "<service xmlns:ohr='http://www.terracotta.org/config/offheap-resource' id=\"resources\">"
+          + "<ohr:offheap-resources>"
+          + "<ohr:resource name=\"primary-server-resource\" unit=\"MB\">64</ohr:resource>"
+          + "</ohr:offheap-resources>" +
+          "</service>\n";
+
+  @Rule
+  public final TestName testName = new TestName();
+
+  @Rule
+  public final Cluster cluster =
+      new BasicExternalCluster(new File("build/cluster"), 1, Collections.<File>emptyList(), "", RESOURCE_CONFIG);
+
+  @Before
+  public void waitForActive() throws Exception {
+    cluster.getClusterControl().waitForActive();
+  }
+
+  /**
+   * Tests if {@link CacheManager#close()} blocks if the client/server connection is disconnected.
+   */
+  @Test
+  public void testTerminationBeforeCacheManagerClose() throws Exception {
+    CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder =
+        CacheManagerBuilder.newCacheManagerBuilder()
+        .with(ClusteringServiceConfigurationBuilder.cluster(cluster.getConnectionURI().resolve("/MyCacheManagerName?auto-create"))
+            .defaultServerResource("primary-server-resource"));
+    PersistentCacheManager cacheManager = clusteredCacheManagerBuilder.build(false);
+    cacheManager.init();
+
+    cluster.getClusterControl().tearDown();
+
+    Future<Void> future = interruptAfter(2, TimeUnit.SECONDS);
+    try {
+      cacheManager.close();
+    } finally {
+      future.cancel(true);
+    }
+
+    // TODO: Add assertion for successful CacheManager.init() following cluster restart (https://github.com/Terracotta-OSS/galvan/issues/30)
```

Comment:
This test should be amended once functionality suggested by Terracotta-OSS/galvan#30 is available.